### DISCUSSION TITLE
Invalid `prisma.*.create()` invocation

### DISCUSSION THREAD

**Discussion by koreagpa-jayce** [prisma/prisma#7850] [Q&A] [answered]
```
nodejs + express
@prisma/client : 2.25.0
prisma : 2.25.0
QUERY_BATCH_SIZE=800
Single PrismaClient instance (export)
```
```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["selectRelationCount", "orderByAggregateGroup"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

A few hours after the node server starts,
prisma.*.create() is not working (All table).

```
Error:
Invalid `prisma.join.create()` invocation:
  Query createOneJoin is required to return data, but found no record(s).
    at cb (/home/ubuntu/.../node_modules/@prisma/client/runtime/index.js:33820:17)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```
SELECT (O), UPDATE (O), CREATE(X)

**I have two bad solutions**
1. Restart server
2. kill query-engin-debian-openssl-1.1.x (`kill $(ps aux | grep 'prisma' | awk '{print $2}')`)
`ubuntu    9932  0.2  1.5 614388 31048 ?        Sl   10:31   0:06 /home/ubuntu/.../node_modules/.prisma/client/query-engine-debian-openssl-1.1.x --enable-raw-queries --port 44573`

Is there a way to solve this problem?

**Comment by janpio** [comment, marked as answer]
This sounds super weird.

Do you have debug logs from the time this happens? (`DEBUG=*` and possibly query logging)
The database is reachable normally when this happens?
How do CPU and memory look at that time?

**Reply by koreagpa-jayce** [reply]
I'm sorry for the late reply.
I knew what the problem was.

SOLUTION :
Increase the ID length to 30.
(I used 25 length ID - prisma1 )

BEFORE :
`@id @default(cuid()) @db.Char(25)`
AFTER :
`@id @default(cuid()) @db.Char(30)`

sorry and thanks

**Reply by janpio** [reply]
Uh wow. Does that mean that executing this once and it causing and error, you could not run any other queries any more?

**Reply by koreagpa-jayce** [reply]
Only create.
select & update is OK.

The ID length was over 25 when the error occurred. (prisma-query-log)

I guess.
1. Once the ID length over 25,
2. All prisma.create() generate over 25 length ID
3. After kill engine, prisma.create() generate 25 length ID again
(It's just my guess)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
